models/issues/issue.go

```diff
@@ -1346,19 +1343,21 @@ func (opts *IssuesOptions) setupSessionNoLimit(sess *xorm.Session) {
 		sess.And(builder.Eq{"repository.is_archived": opts.IsArchived.IsTrue()})
 	}
 
-	if opts.LabelIDs != nil {
-		var noLabelsIDs []int64
-
+	labelsCount := len(opts.LabelIDs)
+	if labelsCount > 0 {
+		noLabelsIDs := make([]int64, 0, labelsCount)
+		hasNoLabel := false
 		for i, labelID := range opts.LabelIDs {
 			if labelID > 0 {
 				sess.Join("INNER", fmt.Sprintf("issue_label il%d", i),
 					fmt.Sprintf("issue.id = il%[1]d.issue_id AND il%[1]d.label_id = %[2]d", i, labelID))
 			} else if labelID < 0 {
 				// sess.Where("issue.id not in (select issue_id from issue_label where label_id = ?)", -labelID)
```

Comment:
![image](https://user-images.githubusercontent.com/1255041/174706691-bb83be90-9eba-492e-8d64-f4785ca7341e.png)

Reply:
Do you mean this? It will be slower if you don't use select in!

Reply:
let em have some time ... I'll have a look once i got time for it - sorry :/

Reply:
Please ping me after 23.

Reply:
@6543 23rd?

Reply:
This is not right. negative number means this labels should not be contained as conditions.